finality-aleph/src/sync/handler/mod.rs

```diff
@@ -580,6 +588,10 @@ where
                     }
                 }
                 ResponseItem::Header(h) => {
+                    let h = match self.verify_header(h) {
+                        Ok(h) => h,
+                        Err(e) => return (new_highest, Some(e)),
+                    };
                     if self.forest.skippable(&h.id()) {
                         continue;
                     }
```

Comment:
is this order right? Just want to make sure it is.

Reply:
The other way has to be correct and it's a minor optimization, so it would be nice to switch around, unless `UnverifiedHeader` doesn't have `id()`, in which case adding it is probably not worth it.

Reply:
fixed